src/main/java/com/example/wdmsystem/employee/system/UpdateEmployeeDTO.java

```diff
@@ -1,6 +1,8 @@
 package com.example.wdmsystem.employee.system;
 
+import com.example.wdmsystem.merchant.system.Merchant;
+
 /// CreateEmployee is used as a DTO in both creation and updates, separated them as they have different fields
-public record UpdateEmployeeDTO(int merchantId, String firstName, String lastName, EmployeeType employeeType, String username,
+public record UpdateEmployeeDTO(Merchant merchant, String firstName, String lastName, EmployeeType employeeType, String username,
```

Comment:
The dto can still just send the id, the merchant can be fetched in the service method. I think if you want to move the employee to another merchant you have to pass all of the merchant details.